typescript/sdk/src/token/deploy.ts

```diff
@@ -83,58 +83,69 @@ abstract class TokenDeployer<
     multiProvider: MultiProvider,
     configMap: WarpRouteDeployConfig,
   ): Promise<TokenMetadata | undefined> {
-    for (const [chain, config] of Object.entries(configMap)) {
-      if (isTokenMetadata(config)) {
-        return config;
-      }
+    try {
+      for (const [chain, config] of Object.entries(configMap)) {
+        if (isTokenMetadata(config)) {
+          return config;
+        }
 
-      if (isNativeConfig(config)) {
-        const nativeToken = multiProvider.getChainMetadata(chain).nativeToken;
-        if (nativeToken) {
-          return { totalSupply: 0, ...nativeToken };
+        if (isNativeConfig(config)) {
+          const nativeToken = multiProvider.getChainMetadata(chain).nativeToken;
+          if (nativeToken) {
+            return { totalSupply: 0, ...nativeToken };
+          }
         }
-      }
 
-      if (isCollateralConfig(config)) {
-        const provider = multiProvider.getProvider(chain);
-
-        if (config.isNft) {
-          const erc721 = ERC721Enumerable__factory.connect(
-            config.token,
-            provider,
-          );
-          const [name, symbol, totalSupply] = await Promise.all([
-            erc721.name(),
-            erc721.symbol(),
-            erc721.totalSupply(),
+        if (isCollateralConfig(config)) {
+          const provider = multiProvider.getProvider(chain);
+
+          if (config.isNft) {
+            const erc721 = ERC721Enumerable__factory.connect(
+              config.token,
+              provider,
+            );
+            const [name, symbol, totalSupply] = await Promise.all([
+              erc721.name(),
+              erc721.symbol(),
+              erc721.totalSupply(),
+            ]);
+            return {
+              name,
+              symbol,
+              totalSupply: totalSupply.toString(),
+            };
+          }
+
+          const erc20 = ERC20__factory.connect(config.token, provider);
+          const [name, symbol, totalSupply, decimals] = await Promise.all([
+            erc20.name(),
+            erc20.symbol(),
+            erc20.totalSupply(),
+            erc20.decimals(),
           ]);
+
           return {
             name,
             symbol,
             totalSupply: totalSupply.toString(),
+            decimals,
           };
         }
-
-        const erc20 = ERC20__factory.connect(config.token, provider);
-        const [name, symbol, totalSupply, decimals] = await Promise.all([
-          erc20.name(),
-          erc20.symbol(),
-          erc20.totalSupply(),
-          erc20.decimals(),
-        ]);
-
-        return { name, symbol, totalSupply: totalSupply.toString(), decimals };
       }
-    }
 
-    return undefined;
+      return undefined;
+    } catch (error) {
```

Comment:
It may have been intentional to let this error propagate before. Are you certain the deployment should proceed even if we can't derive metadata in the way we expected to? Also these changes seem unrelated to SmartProvider and should preferably be in a separate PR

Reply:
heavy agree with this, this is inconsistent now with how the schema assesses whether we have "sufficient" metadata to deploy

Reply:
Yes sorry, re-throwing the error